lib/assert.js

```diff
@@ -110,15 +111,23 @@ function fail(actual, expected, message, operator, stackStartFn) {
       );
     }
     if (argsLen === 2)
-      operator = '!=';
+      operator = defaultOp;
   }
 
   if (message instanceof Error) throw message;
 
+  if (operator === undefined && message) {
+    if (message) {
+      operator = 'fail';
+    } else {
+      operator = defaultOp;
+    }
+  }
```

Comment:
This should likely be changed to:

```js
if (!operator) {
  if (message) {
    operator = 'fail';
  } else {
    operator = '!=';
  }
}
```

Otherwise cases where we have at least four arguments and `operator` is set to `undefined` and `message` is a falsy value, the default operator is not chosen and that seems wrong.